### ISSUE TITLE
Do you intend to submit to CRAN?

### ISSUE THREAD

**Issue by elbamos** [MangoTheCat/visualTest#18]
A good way of testing packages that produce plots, would be very helpful.

**Comment by gaborcsardi** [comment]
Yes, but I think it needs to be improved. Right now it is too forgiving, and probably another comparison algorithm is needed for proper image testing.

**Comment by tvatter** [comment]
bump :)

**Comment by gaborcsardi** [comment]
I am not responsible for this any more, so I don't know. The vdiffr package might be appropriate for you: https://r-pkg.org/pkg/vdiffr

**Comment by tvatter** [comment]
I looked at `vdiffr` and it does not solve exactly the problem that I have, because it can only test for exact equality of pictures. I use `visualTest` in the context of a data science course to automatically grade students' submissions. For instance, I ask them to reproduce a figure by giving them the steps, and I check whether their figure is close enough to the solution. The issue is that `vdiffr` is too strict, while `visualTest` has the useful threshold.

**Comment by gaborcsardi** [comment]
Right. That's a pretty good use case. There is this issue, though, which limits the usefulness of this package greatly: https://github.com/MangoTheCat/visualTest/issues/19
